File: src/models/GreetingModel.ts
import { invoke } from "@tauri-apps/api/core";

export interface GreetingModel {
  name: string;
  message: string;
}

export async function requestGreeting(name: string): Promise<GreetingModel> {
  const message = await invoke<string>("greet", { name });

  return {
    name,
    message,
  };
}

File: src-tauri/src/models/greeting_model.rs
pub struct GreetingModel {
    name: String,
}

impl GreetingModel {
    pub fn new(name: &str) -> Self {
        Self {
            name: name.to_string(),
        }
    }

    pub fn name(&self) -> &str {
        &self.name
    }
}
